ecosystem/node-checker/src/metric_evaluator/consensus/proposals_evaluator.rs

```diff
@@ -0,0 +1,214 @@
+    }
+
+    fn build_evaluation(
+        &self,
+        previous_proposals_count: u64,
+        latest_proposals_count: u64,
+    ) -> EvaluationResult {
+        // We convert to i64 to avoid potential overflow if somehow the count somehow went backwards.
+        let progress = latest_proposals_count as i64 - previous_proposals_count as i64;
+        match progress {
+            progress if (progress == 0) => {
+                EvaluationResult {
+                    headline: "Proposals count is not progressing".to_string(),
+                    score: 50,
+                    explanation: "Successfully pulled metrics from target node twice, but the proposal count isn't progressing.".to_string(),
+                    source: CONSENSUS_EVALUATOR_SOURCE.to_string(),
+                    links: vec![],
+              }
+            }
+            progress if (progress < 0) => {
+                EvaluationResult {
+                    headline: "Proposals count went backwards!".to_string(),
+                    score: 0,
+                    explanation: "Successfully pulled metrics from target node twice, but the second time the proposals count went backwards!".to_string(),
+                    source: CONSENSUS_EVALUATOR_SOURCE.to_string(),
+                    links: vec![],
+                }
+            }
+            _wildcard => {
+                EvaluationResult {
+                    headline: "Proposals count is increasing".to_string(),
+                    score: 100,
+                    explanation: "Successfully pulled metrics from target node twice and saw that proposals count is increasing".to_string(),
+                    source: CONSENSUS_EVALUATOR_SOURCE.to_string(),
+                    links: vec![],
+                }
+            }
+        }
+    }
+}
+
+impl MetricsEvaluator for ConsensusProposalsEvaluator {
+    /// Assert that the proposals count is increasing on the target node.
+    fn evaluate_metrics(
+        &self,
+        _previous_baseline_metrics: &PrometheusScrape,
+        previous_target_metrics: &PrometheusScrape,
+        _latest_baseline_metrics: &PrometheusScrape,
+        latest_target_metrics: &PrometheusScrape,
+    ) -> Result<Vec<EvaluationResult>, MetricsEvaluatorError> {
+        let mut evaluation_results = vec![];
+
+        // Get previous proposals count from the target node.
+        let previous_proposals_count =
+            match self.get_proposals_count(previous_target_metrics, "first") {
+                GetMetricResult::Present(metric) => Some(metric),
+                GetMetricResult::Missing(evaluation_result) => {
+                    evaluation_results.push(evaluation_result);
+                    None
+                }
+            };
+
+        // Get the latest proposals count from the target node.
+        let latest_proposals_count = match self.get_proposals_count(latest_target_metrics, "second")
```

Comment:
nit: this could be a helper (the code is very similar to above and the state sync evaluator below)